Failed to compile diff_drive_controller on Fedora galactic or Foxy

**Describe the bug**
diff_drive_controller fails to compile on Fedora 34 using either Galactic or Foxy. When trying to compile webots_ros2.

**To Reproduce**
Steps to reproduce the behaviour:
1. boot Fedora 34
2. install ROS Foxy
3. try to compile webots_ros2 (they use ros2_control as dependency)
4. See error

**Expected behaviour**
diff_drive_controller being compiled without error

**Screenshots**
```
--- stderr: diff_drive_controller                                                                                                                                                                                                      
In file included from /home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp:18:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:37:41: fout: expected ‘)’ before ‘rolling_window_size’
   37 |   explicit RollingMeanAccumulator(size_t rolling_window_size)
      |                                  ~      ^~~~~~~~~~~~~~~~~~~~
      |                                         )
In file included from /home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp:18:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:61:3: fout: ‘size_t’ does not name a type
   61 |   size_t next_insert_;
      |   ^~~~~~
In file included from /home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp:18:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:24:1: note: ‘size_t’ is defined in header ‘<cstddef>’; did you forget to ‘#include <cstddef>’?
   23 | #include <vector>
  +++ |+#include <cstddef>
   24 | 
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp: In memberfunctie ‘void diff_drive_controller::RollingMeanAccumulator<T>::accumulate(T)’:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:44:21: fout: ‘next_insert_’ was not declared in this scope
   44 |     sum_ -= buffer_[next_insert_];
      |                     ^~~~~~~~~~~~
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp: In memberfunctie ‘T diff_drive_controller::RollingMeanAccumulator<T>::getRollingMean() const’:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:54:5: fout: ‘size_t’ was not declared in this scope; did you mean ‘std::size_t’?
   54 |     size_t valid_data_count = buffer_filled_ * buffer_.size() + !buffer_filled_ * next_insert_;
      |     ^~~~~~
      |     std::size_t
In file included from /usr/include/c++/11/cassert:43,
                 from /home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:22,
                 from /home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp:18:
/usr/include/c++/11/x86_64-redhat-linux/bits/c++config.h:2426:33: note: ‘std::size_t’ declared here
 2426 |   typedef __SIZE_TYPE__         size_t;
      |                                 ^~~~~~
In file included from /usr/include/c++/11/cassert:44,
                 from /home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:22,
                 from /home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp:18:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:55:12: fout: ‘valid_data_count’ was not declared in this scope
   55 |     assert(valid_data_count > 0);
      |            ^~~~~~~~~~~~~~~~
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp: In memberfunctie ‘virtual void TestAccumulator_test_accumulator_Test::TestBody()’:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp:27:64: fout: no matching function for call to ‘diff_drive_controller::RollingMeanAccumulator<double>::RollingMeanAccumulator(int)’
   27 |   diff_drive_controller::RollingMeanAccumulator<double> accum(4);
      |                                                                ^
In file included from /home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp:18:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note: candidate: ‘diff_drive_controller::RollingMeanAccumulator<double>::RollingMeanAccumulator()’
   34 | class RollingMeanAccumulator
      |       ^~~~~~~~~~~~~~~~~~~~~~
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note:   candidate expects 0 arguments, 1 provided
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note: candidate: ‘diff_drive_controller::RollingMeanAccumulator<double>::RollingMeanAccumulator(const diff_drive_controller::RollingMeanAccumulator<double>&)’
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note:   no known conversion for argument 1 from ‘int’ to ‘const diff_drive_controller::RollingMeanAccumulator<double>&’
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note: candidate: ‘diff_drive_controller::RollingMeanAccumulator<double>::RollingMeanAccumulator(diff_drive_controller::RollingMeanAccumulator<double>&&)’
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note:   no known conversion for argument 1 from ‘int’ to ‘diff_drive_controller::RollingMeanAccumulator<double>&&’
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp: In memberfunctie ‘virtual void TestAccumulator_spam_accumulator_Test::TestBody()’:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp:52:65: fout: no matching function for call to ‘diff_drive_controller::RollingMeanAccumulator<double>::RollingMeanAccumulator(int)’
   52 |   diff_drive_controller::RollingMeanAccumulator<double> accum(10);
      |                                                                 ^
In file included from /home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/test/test_accumulator.cpp:18:
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note: candidate: ‘diff_drive_controller::RollingMeanAccumulator<double>::RollingMeanAccumulator()’
   34 | class RollingMeanAccumulator
      |       ^~~~~~~~~~~~~~~~~~~~~~
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note:   candidate expects 0 arguments, 1 provided
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note: candidate: ‘diff_drive_controller::RollingMeanAccumulator<double>::RollingMeanAccumulator(const diff_drive_controller::RollingMeanAccumulator<double>&)’
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note:   no known conversion for argument 1 from ‘int’ to ‘const diff_drive_controller::RollingMeanAccumulator<double>&’
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note: candidate: ‘diff_drive_controller::RollingMeanAccumulator<double>::RollingMeanAccumulator(diff_drive_controller::RollingMeanAccumulator<double>&&)’
/home/noeel/dev_ws/webots_foxy/src/ros2_controllers/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp:34:7: note:   no known conversion for argument 1 from ‘int’ to ‘diff_drive_controller::RollingMeanAccumulator<double>&&’
gmake[2]: *** [CMakeFiles/test_accumulator.dir/build.make:76: CMakeFiles/test_accumulator.dir/test/test_accumulator.cpp.o] Fout 1
gmake[1]: *** [CMakeFiles/Makefile2:129: CMakeFiles/test_accumulator.dir/all] Fout 2
gmake[1]: *** Wachten op onvoltooide taken...
gmake: *** [Makefile:146: all] Fout 2
---
Failed   <<< diff_drive_controller [22.6s, exited with code 2]

```

**Environment (please complete the following information):**
 - OS: Fedora 34
 - Version Foxy
 - gcc: 11.2.1
 - glibc 2.33
 - arch: x86-64
 - webots_ros2: foxy

**Additional context**
I already made an issue about this on the webots_ros2 repo https://github.com/cyberbotics/webots_ros2/issues/302, but I think that the error might lie outside their code.

--- a/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp
+++ b/diff_drive_controller/include/diff_drive_controller/rolling_mean_accumulator.hpp
@@ -20,8 +20,8 @@
 #define DIFF_DRIVE_CONTROLLER__ROLLING_MEAN_ACCUMULATOR_HPP_
 
 #include <cassert>
-#include <vector>
 #include <cstdio>
+#include <vector>
 
 namespace diff_drive_controller
 {

